internal/services/referrals.go

```diff
@@ -52,72 +59,148 @@ func NewReferralBonusService(
 	}
 }
 
+const level2Weeks = 4
+
 // CollectReferrals returns address pairs for referrals completed in the given week.
 // These will come from referees who are earning for the first time and have a referrer
 // attached to their account.
 func (c *ReferralsClient) CollectReferrals(ctx context.Context, issuanceWeek int) (Referrals, error) {
 	var refs Referrals
 
-	var res []struct {
-		UserID string `boil:"r1.user_id"`
-	}
+	logger := c.Logger.With().Int("issuanceWeek", issuanceWeek).Logger()
 
-	var rCols = models.RewardColumns
-
-	err := queries.Raw(
-		"SELECT DISTINCT ON (r1."+rCols.UserEthereumAddress+")"+
-			" r1."+rCols.UserID+
-			" FROM "+models.TableNames.Rewards+" r1"+
-			" LEFT OUTER JOIN "+models.TableNames.Rewards+" r2 ON r1."+rCols.UserEthereumAddress+" = r2."+rCols.UserEthereumAddress+" AND r2."+rCols.IssuanceWeekID+" < $1"+
-			" LEFT JOIN "+models.TableNames.Vins+" v on r1."+rCols.UserDeviceTokenID+" = v."+models.VinColumns.FirstEarningTokenID+
-			" WHERE r1."+rCols.IssuanceWeekID+" = $1 AND r1."+rCols.UserEthereumAddress+" IS NOT NULL AND r2."+rCols.UserDeviceID+" IS NULL"+
-			" AND v."+models.VinColumns.FirstEarningWeek+" = $1"+
-			" ORDER BY r1."+rCols.UserEthereumAddress+", r1."+rCols.UserID,
-		issuanceWeek,
-	).Bind(ctx, c.TransferService.db.DBS().Reader, &res)
+	vehicleNFTsHittingLevel2, err := models.Rewards(
+		models.RewardWhere.IssuanceWeekID.EQ(issuanceWeek),
+		models.RewardWhere.ConnectionStreak.EQ(level2Weeks),
+		models.RewardWhere.DisconnectionStreak.EQ(0),
+	).All(ctx, c.TransferService.db.DBS().Reader)
```

Comment:
should we update the comment above to explain the new logic (only earning after reaching level 2?)

Reply:
Will do.